components/mediation/tasks/org.wso2.micro.integrator.ntask.core/src/main/java/org/wso2/micro/integrator/ntask/coordination/task/db/connector/CoordinatedTaskDBConnector.java

```diff
@@ -0,0 +1,362 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.micro.integrator.ntask.coordination.task.db.connector;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.micro.integrator.ntask.coordination.task.ConstantUtil;
+import org.wso2.micro.integrator.ntask.coordination.task.CoordinateTaskRunTimeException;
+import org.wso2.micro.integrator.ntask.coordination.task.CoordinatedTask;
+import org.wso2.micro.integrator.ntask.coordination.task.db.TaskQueryHelper;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.sql.DataSource;
+
+/**
+ * The connector class which deals with underlying coordinated task table.
+ */
+public class CoordinatedTaskDBConnector {
```

Comment:
I feel like it's better if we can rename this class to something generic like 'DBConnector' since this is just a dumb db query executor. This being inside the package says that it has something to do with tasks.

Reply:
If so we have give to pass the queries also. It is more coupled with TASK_TABLE , thats why I named it like it . wdyt

Reply:
But it's inside the task package, so the DBConnector inside the task package gives you the idea that it deals with the tasks table, does it not?